presto-main/src/main/java/io/prestosql/type/UuidOperators.java

```diff
@@ -60,6 +67,123 @@ public static Slice uuid()
+    }
+
+    @Description("Returns the URL namespace constant for UUIDs")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_ns_url()
+    {
+        String nameSpaceUrl = "6ba7b811-9dad-11d1-80b4-00c04fd430c8";
+        java.util.UUID nsUrlUuid = java.util.UUID.fromString(nameSpaceUrl);
+        return wrappedLongArray(nsUrlUuid.getMostSignificantBits(), nsUrlUuid.getLeastSignificantBits());
+    }
+
+    @Description("Returns the ISO object identifier (OID) namespace constant for UUIDs")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_ns_oid()
+    {
+        String nameSpaceOid = "6ba7b812-9dad-11d1-80b4-00c04fd430c8";
+        java.util.UUID nsOidUuid = java.util.UUID.fromString(nameSpaceOid);
+        return wrappedLongArray(nsOidUuid.getMostSignificantBits(), nsOidUuid.getLeastSignificantBits());
+    }
+
+    @Description("Returns the X.500 distinguished name (DN) namespace constant for UUIDs")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_ns_x500()
+    {
+        String nameSpaceX500 = "6ba7b814-9dad-11d1-80b4-00c04fd430c8";
+        java.util.UUID nsX500Uuid = java.util.UUID.fromString(nameSpaceX500);
+        return wrappedLongArray(nsX500Uuid.getMostSignificantBits(), nsX500Uuid.getLeastSignificantBits());
+    }
+
+    @Description("Generates a deterministic (type 3) UUID using the specified namespace UUID and input name (using MD5 hash)")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_v3(@SqlType(StandardTypes.UUID) Slice nameSpaceUuid, @SqlType(StandardTypes.VARCHAR) Slice name)
+    {
+        byte[] value = name.getBytes();
+
+        java.util.UUID nameUuid = new java.util.UUID(nameSpaceUuid.getLong(0), nameSpaceUuid.getLong(SIZE_OF_LONG));
+        String uuidHexString = nameUuid.toString().replace("-", "");
+
+        byte[] nameSpaceBytes = new byte[16];
+        for (int i = 0; i < 16; i++) {
+            int firstDigit = Character.digit(uuidHexString.substring(i * 2, i * 2 + 2).charAt(0), 16);
+            int secondDigit = Character.digit(uuidHexString.substring(i * 2, i * 2 + 2).charAt(1), 16);
+            byte b = (byte) ((firstDigit << 4) + secondDigit);
+            nameSpaceBytes[i] = b;
+        }
+
+        java.util.UUID uuid = nameUUIDFromBytes(Bytes.concat(nameSpaceBytes, value));
+        return wrappedLongArray(uuid.getMostSignificantBits(), uuid.getLeastSignificantBits());
+    }
+
+    @Description("Generates a deterministic (type 5) UUID using the specified namespace UUID and input name (using SHA hash)")
+    @ScalarFunction
+    @SqlType(StandardTypes.UUID)
+    public static Slice uuid_v5(@SqlType(StandardTypes.UUID) Slice nameSpaceUuid, @SqlType(StandardTypes.VARCHAR) Slice name)
+    {
+        byte[] value = name.getBytes();
+
+        java.util.UUID nameUuid = new java.util.UUID(nameSpaceUuid.getLong(0), nameSpaceUuid.getLong(SIZE_OF_LONG));
```

Comment:
```java
        java.util.UUID nameUuid = new java.util.UUID(nameSpaceUuid.getLong(0), nameSpaceUuid.getLong(SIZE_OF_LONG));
        String uuidHexString = nameUuid.toString().replace("-", "");

        byte[] nameSpaceBytes = new byte[16];
        for (int i = 0; i < 16; i++) {
            int firstDigit = Character.digit(uuidHexString.substring(i * 2, i * 2 + 2).charAt(0), 16);
            int secondDigit = Character.digit(uuidHexString.substring(i * 2, i * 2 + 2).charAt(1), 16);
            byte b = (byte) ((firstDigit << 4) + secondDigit);
            nameSpaceBytes[i] = b;
        }
```
can be extracted as helper function `private static byte[] getUuidBytes(Slice uuid)`

Reply:
Okay, I was also wondering the same but didn't know where to put it. 

Should I put it in the `UuidOperators.java` file itself?

Reply:
Yeah - I would do so :)